fdbserver/RatekeeperInterface.h

```diff
@@ -79,28 +79,37 @@ struct TransactionCommitCostEstimation {
 	int numWrite = 0;
 	int numAtomicWrite = 0;
 	int numClear = 0;
-	int numClearShards = 0;
-	uint64_t bytesWrite = 0;
-	uint64_t bytesAtomicWrite = 0;
-	uint64_t bytesClearEst = 0;
+	uint64_t costWrite = 0;
+	uint64_t costAtomicWrite = 0;
+	uint64_t costClearEst = 0;
+
+	uint64_t getCostSum() const { return costClearEst + costAtomicWrite + costWrite; }
+	int getOpsSum() const { return numWrite + numAtomicWrite + numClear; }
 
 	template <class Ar>
 	void serialize(Ar& ar) {
-		serializer(ar, bytesWrite, bytesClearEst, bytesAtomicWrite, numWrite, numAtomicWrite, numClear, numClearShards);
+		serializer(ar, costWrite, costClearEst, costAtomicWrite, numWrite, numAtomicWrite, numClear);
 	}
 
 	TransactionCommitCostEstimation& operator+=(const TransactionCommitCostEstimation& other) {
 		numWrite += other.numWrite;
 		numAtomicWrite += other.numAtomicWrite;
 		numClear += other.numClear;
-		bytesWrite += other.bytesWrite;
-		bytesAtomicWrite += other.numAtomicWrite;
-		numClearShards += other.numClearShards;
-		bytesClearEst += other.bytesClearEst;
+		costWrite += other.costWrite;
+		costAtomicWrite += other.numAtomicWrite;
+		costClearEst += other.costClearEst;
 		return *this;
 	}
 };
-
+struct ClientTrCommitCostEstimation {
+	int opsCount = 0;
+	uint64_t writeCosts = 0;
+	std::map<int, uint64_t> clearIdxCosts;
```

Comment:
A `std::deque<std::pair<int, uint64_t>>` is probably a more efficient data structure for this field, because we never need random access. Currently, we can't serialize `std::deque`, but this functionality could be added to `serialize.h`. It would be very similar to the existing serialization/deserialization implementations for other standard containers.

Reply:
`mutationNum` is always increasing from 0 to `pMutations->size()-1` in `assignMutationsToStorageServers`, so you only need to perform a linear scan of `clearIdxCosts` in this case.

Reply:
Not every index between [0, pMutations->size()-1] will appear in cleearIdxCosts. Only those are sampled will.